### PR TITLE
No errors for non-virtual members overrides

### PR DESCRIPTION
**Pull request by edgarfgp**
Fixes https://github.com/dotnet/fsharp/issues/11456

Reuses error ```855, tcNoMemberFoundForOverride,"No abstract or interface member was found that corresponds to this override"``` for consistency.

But a clearer message would be 
```'C.M2': cannot override inherited member 'B.M2' because it is not marked virtual, abstract, or override."```
- This will need to be a new error message 
- Virtual is not used in F# and in this case refers to C#. So might be confusing
- It will require some string manipulation to get the type name, member name from the declaring type and the inherited type

### CODE REVIEW

**Comment by vzarytovskii** [comment]
If this was compiling before, this will need to go under la nguage version flag.

**Comment by edgarfgp** [comment]
@vzarytovskii Any ideal why the CI is failing with ```error FS0073: (NETCORE_ENGINEERING_TELEMETRY=Build) internal error: not a .NET type definition (Failure)```

I will add a preview language flag

**Comment by vzarytovskii** [comment]
> @vzarytovskii Vlad Zarytovskii FTE Any idea why the CI is failing with `error FS0073: (NETCORE_ENGINEERING_TELEMETRY=Build) internal error: not a .NET type definition (Failure)`
> 
Erm, never seen it before, let me look if it's related to changes.

**Comment by vzarytovskii** [comment]
So, the error thrown in the Proto compiler, so it's likely related to the change:
```fsharp
/// Get the Abstract IL scope, nesting and metadata for this 
/// type definition, assuming it is backed by Abstract IL metadata.
member x.ILTyconInfo = match x.TypeReprInfo with | TILObjectRepr data -> data | _ -> failwith "not a .NET type definition"
```

**Comment by edgarfgp** [comment]
> So, the error thrown in the Proto compiler, so it's likely related to the change:
> 
> ```fsharp
> /// Get the Abstract IL scope, nesting and metadata for this 
> /// type definition, assuming it is backed by Abstract IL metadata.
> member x.ILTyconInfo = match x.TypeReprInfo with | TILObjectRepr data -> data | _ -> failwith "not a .NET type definition"
> ```

Thanks, will look into it.

**Comment by edgarfgp** [comment]
@vzarytovskii it is green now 😀

File: src/Compiler/Checking/CheckExpressions.fs [hunk 1]
```diff
@@ -10997,7 +10997,21 @@ and ApplyAbstractSlotInference (cenv: cenv) (envinner: TcEnv) (argsAndRetTy, m,
                      | _ -> [] // check that method to override is sealed is located at CheckOverridesAreAllUsedOnce (typrelns.fs)
                       // We hit this case when it is ambiguous which abstract method is being implemented.
 
-
+             if g.langVersion.SupportsFeature(LanguageFeature.ErrorForNonVirtualMembersOverrides) then
+                 // Checks if the declaring type inherits from a base class and is not FSharpObjModelTy
+                 // Raises an error if we try to override an non virtual member with the same name in both
+                 match baseValOpt with
+                 | Some ttype when not(isFSharpObjModelTy g ttype.Type) ->
+                    match ttype.Type with
```
Comment (vzarytovskii) [inline src/Compiler/Checking/CheckExpressions.fs:11005]: We almost never want to match on `TType`, should use `stripTyEqns` instead

**Review by vzarytovskii — CHANGES_REQUESTED** [review]

File: src/Compiler/Checking/CheckExpressions.fs (hunk 1 above)
Comment (edgarfgp) [reply src/Compiler/Checking/CheckExpressions.fs]: Updated. Thanks for the review

**Review by vzarytovskii — APPROVED** [review]

**Review by T-Gro — APPROVED** [review]

**Pull request merged**